com.unity.ml-agents.extensions/Runtime/Sensors/PoseExtractor.cs

```diff
@@ -21,45 +21,113 @@ public abstract class PoseExtractor
         Vector3[] m_ModelSpaceLinearVelocities;
         Vector3[] m_LocalSpaceLinearVelocities;
 
+        bool[] m_PoseEnabled;
+
 
         /// <summary>
         /// Read access to the model space transforms.
         /// </summary>
-        public IList<Pose> ModelSpacePoses
+        public IEnumerable<Pose> GetModelSpacePoses()
         {
-            get { return m_ModelSpacePoses;  }
+            if (m_ModelSpacePoses == null)
+            {
+                yield break;
+            }
+
+            for (var i = 0; i < m_ModelSpacePoses.Length; i++)
+            {
+                if (m_PoseEnabled[i])
+                {
+                    yield return m_ModelSpacePoses[i];
+                }
+            }
         }
 
         /// <summary>
         /// Read access to the local space transforms.
         /// </summary>
-        public IList<Pose> LocalSpacePoses
+        public IEnumerable<Pose> GetLocalSpacePoses()
         {
-            get { return m_LocalSpacePoses;  }
+            if (m_LocalSpacePoses == null)
+            {
+                yield break;
+            }
+
+            for (var i = 0; i < m_LocalSpacePoses.Length; i++)
+            {
+                if (m_PoseEnabled[i])
+                {
+                    yield return m_LocalSpacePoses[i];
+                }
+            }
         }
 
         /// <summary>
         /// Read access to the model space linear velocities.
         /// </summary>
-        public IList<Vector3> ModelSpaceVelocities
+        public IEnumerable<Vector3> GetModelSpaceVelocities()
```

Comment:
is this to make `foreach` loops easier to write?

Reply:
Followed up offline. Renamed to GetEnabledModelSpaceVelocities (and similar for friends)